CMakeLists.txt

```diff
@@ -83,14 +171,13 @@ macro(nbl_adjust_flags)
 	option(NBL_GCC_SANITIZE_THREAD OFF) # @Anastazluk did MSVC finally include the thread sanitizer?
 
 	if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
-		if (NOT NBL_BUILD_CEGUI)
-			add_compile_options(-fno-exceptions)
+		
+		if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+			add_compile_options("$<$<CONFIG:DEBUG>:-g>")
```

Comment:
you are not adding any debugging for CLANG and you've broken `RelWithDebInfo` on GCC